lib/datadog/error_tracking/collector.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require_relative 'ext'
+
+module Datadog
+  module ErrorTracking
+    # The Collector is in charge, for a SpanOperation of storing the span events
+    # created when an error is handled. Each SpanOperation has a Collector as soon
+    # as a span event is created and the Collector has the same life time as the SpanOp.
+    #
+    # If an error is handled then rethrown, the SpanEvent corresponding to the error
+    # will be deleted. That is why we do not add directly the SpanEvent to the SpanOp.
+    #
+    # @api private
+    class Collector
+      SPAN_EVENTS_LIMIT = 100
+      LOCK = Mutex.new
+      # Proc called when the span_operation :after_stop event is published
+      def self.after_stop
+        @after_stop ||= proc do |span_op, error|
+          collector = span_op.collector
+          # if an error exited the scope of the span, we delete the corresponding SpanEvent.
+          collector.on_error(span_op, error) unless error.nil?
+
+          span_events = collector.get_span_events
+          span_op.span_events.concat(span_events) if span_events
+        end
+      end
+
+      def initialize
+        @span_event_per_error = {}
+      end
+
+      def add_span_event(span_op, error, span_event)
```

Comment:
according to the name span arguments have priority, hence

```suggestion
      def add_span_event(span_op, span_event, error)
```

Reply:
addressed in e5fe33bee6f4993593a80e133f14416eee497861